server/utils/user/user.utils.ts

```diff
@@ -1,15 +1,37 @@
 import { Context } from "@/server/context";
+import { PrismaClient } from "@prisma/client";
+import { ITXClientDenyList } from "@prisma/client/runtime/library";
 
-export function getUserConnectOrCreate(
-  ctx: Context,
-) {
+export async function getUserConnectOrCreate(ctx: Context) {
   if (!ctx.user) {
     throw new Error("Missing `ctx.user`");
   }
 
+  // First check if the user profile exists
+  const existingProfile = await ctx.prisma.userProfile.findUnique({
+    select: {
+      supId: true,
+    },
+    where: {
+      supId: ctx.user.id,
+    },
+  });
+  
+  // If user profile exists, return a connect object
+  if (existingProfile) {
+    return {
+      connect: {
+        supId: ctx.user.id,
+      },
+    };
+  }
+  
+  // If not found, return a create object with minimal required fields
   return {
```

Comment:
This should never happen. The function was incorrectly named. It should not be possible to indirectly create an user by creating a different entity.

Reply:
renamed to getUserProfile and now throws an error if it cannot find the profile.